File: app/src/main/java/com/example/sendserverdemo/BaseResponse.java
package com.example.sendserverdemo;

public class BaseResponse<T> {
    public boolean isSuccess;

    public int errorCode;

    public String errorMsg;

    public T data;

    @Override
    public String toString() {
        return "BaseData{" +
                "isSuccess=" + isSuccess +
                ", errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}';
    }
}

File: retrofit/src/main/java/com/kcst/retrofit/base/BaseResponse.kt
package com.kcst.retrofit.base

abstract class BaseResponse<T> {
    var isSuccess = false
    var errorCode = 0
    var errorMsg: String? = null

    abstract var data: T?
    override fun toString(): String {
        return "BaseResponse(isSuccess=$isSuccess, errorCode=$errorCode, errorMsg=$errorMsg, data=$data)"
    }
}